python/ray/data/impl/block_builder.py

```diff
@@ -1,11 +1,35 @@
 from typing import Generic
 
-from ray.data.block import Block, T
+from ray.data.block import Block, T, U
 
 
 class BlockBuilder(Generic[T]):
     """A builder class for blocks."""
 
+    @staticmethod
+    def for_block_type(block_type: U) -> "BlockBuilder[T]":
+        import pyarrow
+        import pandas
```

Comment:
This should delegate to accessor.for_block right? I don't think we need to take in raw block types, it should be enough to support BlockBuilder.for_block(block).

Reply:
Yeah that also works.